docs/concepts/registries.md

```diff
@@ -1,200 +1,232 @@
+@SubscribeEvent
+public void register(RegisterEvent event) {
+    event.register(
+            // This is the registry key of the registry.
+            // Get these from Registries for vanilla registries, or from NeoForgeRegistries.Keys for NeoForge registries.
+            Registries.BLOCKS,
+            // Register your objects here.
+            registry -> {
+                registry.register(new ResourceLocation(MODID, "example_block_1"), new Block(...));
+                registry.register(new ResourceLocation(MODID, "example_block_2"), new Block(...));
+                registry.register(new ResourceLocation(MODID, "example_block_3"), new Block(...));
+            }
+    );
+}
 ```
 
-### Using @ObjectHolder
+## Custom Registries
 
-Registered objects from registries can be injected into the `public static` fields by annotating classes or fields with `@ObjectHolder` and supplying enough information to construct a `ResourceLocation` to identify a specific object in a specific registry.
+Custom registries allow you to specify additional systems that addon mods for your mod may want to plug into. For example, if your mod were to add spells, you could make the spells a registry and thus allow other mods to add spells to your mod, without you having to do anything else. It also allows you to do some things, such as syncing the entries, automatically.
 
-The rules for `@ObjectHolder` are as follows:
+Let's start by creating the [registry key][resourcekey] and the registry itself:
 
-* If the class is annotated with `@ObjectHolder`, its value will be the default namespace for all fields within if not explicitly defined
-* If the class is annotated with `@Mod`, the modid will be the default namespace for all annotated fields within if not explicitly defined
-* A field is considered for injection if:
-  * it has at least the modifiers `public static`;
-  * the **field** is annotated with `@ObjectHolder`, and:
-    * the name value is explicitly defined; and
-    * the registry name value is explicitly defined
-  * _A compile-time exception is thrown if a field does not have a corresponding registry or name._
-* _An exception is thrown if the resulting `ResourceLocation` is incomplete or invalid (non-valid characters in path)_
-* If no other errors or exceptions occur, the field will be injected
-* If all of the above rules do not apply, no action will be taken (and a message may be logged)
+```java
+// We use spells as an example for the registry here, without any details about what a spell actually is (as it doesn't matter).
+// Of course, all mentions of spells can and should be replaced with whatever your registry actually is.
+public static final ResourceKey<Registry<Spell>> SPELL_REGISTRY_KEY = ResourceKey.createRegistryKey(new ResourceLocation("yourmodid", "spells"));
+public static final Registry<YourRegistryContents> SPELL_REGISTRY = new RegistryBuilder<>(SPELL_REGISTRY_KEY)
+        // If you want the registry to sync its values.
```

Comment:
This is misleading. The sync option syncs registry IDs and is also discouraged as you should usually send actual ids instead of int ids. Should also note that synced registries will prevent clients from connecting to servers if they're missing entries present on the other side.

Reply:
```suggestion
// If you want the registry to sync its registry entry IDs for use in networking
```